src/main/java/com/sarality/sync/APIRequestGenerator.java

```diff
@@ -0,0 +1,17 @@
+package com.sarality.sync;
+
+import java.util.List;
+
+/**
+ * generates API Request objects from the source domain data objects
+ *
+ * @author satya@ (Satya Puniani)
+ */
+public interface APIRequestGenerator<T, R> {
+
+  void init(List<T> dataList);
```

Comment:
Can you please add documentation on what these various parameters and methods do.

The fact that what the interface does is not obvious also worries me. Hopefully the documentation fixed this.